source/MissionAction.cpp

```diff
@@ -33,7 +33,16 @@ PARTICULAR PURPOSE.  See the GNU General Public License for more details.
 using namespace std;
 
 namespace {
-	void DoGift(PlayerInfo &player, const Outfit *outfit, int count, UI *ui)
+	void DoGiftShip(PlayerInfo &player, const Ship *modelName, const std::string &name, UI *ui)
+	{
+		player.GiftShip(modelName, name);
+		
+		string message = "The " + name + " was added to your fleet!";
+		if(ui)
+			ui->Push(new Dialog(message));
```

Comment:
This dialog can only ever be one thing, so I wouldn't bother with storing it in a variable. Just pass it directly to the dialog.
```suggestion
		if(ui)
			ui->Push(new Dialog("The " + name + " was added to your fleet!"));
```